src/drivers/magnetometer/bosch/bmm350/BMM350.cpp

```diff
@@ -0,0 +1,757 @@
+				dut_offset_coeff[0] = _mag_comp_vals.dut_offset_coef.offset_x;
+				dut_offset_coeff[1] = _mag_comp_vals.dut_offset_coef.offset_y;
+				dut_offset_coeff[2] = _mag_comp_vals.dut_offset_coef.offset_z;
+
+				dut_sensit_coeff[0] = _mag_comp_vals.dut_sensit_coef.sens_x;
+				dut_sensit_coeff[1] = _mag_comp_vals.dut_sensit_coef.sens_y;
+				dut_sensit_coeff[2] = _mag_comp_vals.dut_sensit_coef.sens_z;
+
+				dut_tcos[0] = _mag_comp_vals.dut_tco.tco_x;
+				dut_tcos[1] = _mag_comp_vals.dut_tco.tco_y;
+				dut_tcos[2] = _mag_comp_vals.dut_tco.tco_z;
+
+				dut_tcss[0] = _mag_comp_vals.dut_tcs.tcs_x;
+				dut_tcss[1] = _mag_comp_vals.dut_tcs.tcs_y;
+				dut_tcss[2] = _mag_comp_vals.dut_tcs.tcs_z;
+
+				for (int idx = 0; idx < 3; idx++) {
+					out_data[idx] *= 1 + dut_sensit_coeff[idx];
+					out_data[idx] += dut_offset_coeff[idx];
+					out_data[idx] += dut_tcos[idx] * (out_data[3] - _mag_comp_vals.dut_t0);
+					out_data[idx] /= 1 + dut_tcss[idx] * (out_data[3] - _mag_comp_vals.dut_t0);
+				}
+
+				cr_ax_comp_x = (out_data[0] - _mag_comp_vals.cross_axis.cross_x_y * out_data[1]) /
+					       (1 - _mag_comp_vals.cross_axis.cross_y_x * _mag_comp_vals.cross_axis.cross_x_y);
+				cr_ax_comp_y = (out_data[1] - _mag_comp_vals.cross_axis.cross_y_x * out_data[0]) /
+					       (1 - _mag_comp_vals.cross_axis.cross_y_x * _mag_comp_vals.cross_axis.cross_x_y);
+				cr_ax_comp_z =
+					(out_data[2] +
+					 (out_data[0] *
+					  (_mag_comp_vals.cross_axis.cross_y_x * _mag_comp_vals.cross_axis.cross_z_y -
+					   _mag_comp_vals.cross_axis.cross_z_x) -
+					  out_data[1] *
+					  (_mag_comp_vals.cross_axis.cross_z_y - _mag_comp_vals.cross_axis.cross_x_y *
+					   _mag_comp_vals.cross_axis.cross_z_x)) /
+					 (1 - _mag_comp_vals.cross_axis.cross_y_x * _mag_comp_vals.cross_axis.cross_x_y));
+
+				out_data[0] = cr_ax_comp_x;
+				out_data[1] = cr_ax_comp_y;
+				out_data[2] = cr_ax_comp_z;
+				_px4_mag.set_error_count(perf_event_count(_bad_read_perf) + perf_event_count(_self_test_failed_perf));
+				_px4_mag.set_temperature(out_data[3]);
+				_px4_mag.update(now, out_data[0], out_data[1], out_data[2]);
+
+			} else {
+				perf_count(_bad_read_perf);
+			}
+		}
+
+		break;
+	}
+}
+
+int BMM350::Configure()
+{
+	PX4_DEBUG("Configuring");
+	uint8_t readData = 0;
+	int ret;
+
+	// Set pad drive
+	ret = RegisterWrite(Register::PAD_CTRL, (_mag_pad_drive & 0x7));
+
+	if (ret != PX4_OK) {
+		PX4_DEBUG("Couldn't set pad drive, defaults to 7");
+	}
+
+	// Set PMU data aggregation
+	uint8_t odr = _mag_odr_mode;
+	uint8_t avg = _mag_avg_mode;
+
+	if (odr == ODR_400HZ && avg >= AVG_2) {
+		avg = AVG_NO_AVG;
+
+	} else if (odr == ODR_200HZ && avg >= AVG_4) {
+		avg = AVG_2;
+
+	} else if (odr == ODR_100HZ && avg >= AVG_8) {
+		avg = AVG_4;
+	}
+
+	uint8_t odr_reg_data = (odr & 0xf);
+	odr_reg_data = ((odr_reg_data & ~(0x30)) | ((avg << 0x4) & 0x30));
+
+	ret = RegisterWrite(Register::PMU_CMD_AGGR_SET, odr_reg_data);
+
+	if (ret != PX4_OK) {
```

Comment:
If writes fails during configure simply return with an error code.
We don't want to have a "half" configured sensor.